sdn_tests/testplans/Stress_Latency_Test_Plan.md

```diff
@@ -0,0 +1,385 @@
+
+    -   Validation:
+         Verify that all the requests have been successful and response messages reflect that.
+
+-   Send a valid request for root with very short timeout value( broken client )
+    -   Condition to test:
+         The connection expected to close before the response received from the server
+
+    -   Validation:
+         Verify that the client closes the connection and gracefully exits.
+
+-   Send a valid request for a single leaf multiple times from a single client
+    -   Condition to test:
+         Set the incremental value for a single leaf over 100 times from single client
+
+    -   Validation:
+         Verify that each request has been fully processed and expected response is received, verify that the state leaf also reflect the configured value
+
+-   Send a valid get requests for a single leaf multiple times from a single client
+    -   Condition to test:
+         Set the incremental value for a single leaf over the 30 minutes period from single client
+
+    -   Validation:
+         Verify that each request has been fully processed and expected response is received, verify that the state leaf also reflect the configured value
+
+# E2E Latency Test Cases
+
+The purpose of the gNMI latency testing is to produce a latency report for various types of the gNMI operations on GPINs platform. The latency report can be viewed through a dashboard. The latency will be measured in milliseconds. The latency measurement will be an internal clock on the test server rather than relying on the gNMI response timestamp. All the Latency measurement tests will use only one guitar cluster( the one which is closest to the DUT )
+
+The end to end latency is measured from the test server in the place where we invoke the gNMI request. The latency measurement will be implemented as a text fixture. All the gNMI based GPINs E2E tests will use the latency test fixture in their test cases to measure the latency for that tests. The fixture measures the latency of the gNMI server end to end irrespective of the test validation of the functionality or the feature under test.
+
+The first few(5) runs are used to establish a baseline benchmark for the GPINs E2E latency report. The applications owners ( responsible for each feature ) will decide the expected latency threshold based on the report. Once the baseline benchmark is established for each gNMI operation, the future runs are going to be compared against the baseline benchmarks. The sandcastle config push latency report can be used as a guideline for the gNMI set operations.
+
+The plan is to implement a latency measurement as part of the Ondatra get/set API so that It will be no-op for the test writer and it will collect the E2E latency for all the GPINS GNMI E2E tests. Ondata will provide project based bindings where the latency measurement for a given operations can be pushed in to the database and dashboard can set up to view the data that are stored in the DB
```

Comment:
Remove "GPINS" on this line. Also, there is a type on this line -- "Ondata" should be Ondatra.